### PR TITLE
adjust for react-router v4. wip

### PR DESCRIPTION
**Pull request by aleksxor**

### CODE REVIEW

**Comment by duvillierA** [comment]
Promising

File: modules/components/AsyncConnect.js [hunk 1]
```diff
@@ -53,16 +55,14 @@ export class AsyncConnect extends Component {
   }
 
   componentWillReceiveProps(nextProps) {
+    const navigated = this.props.location !== nextProps.location;
```
Comment (duvillierA) [inline modules/components/AsyncConnect.js]: if `location` is a router object, you should do a deepDiff to compare objects.

Comment (AVVS) [reply modules/components/AsyncConnect.js]: as far as I'm aware object is changing every time a route changes, so its enough to do a shallow check as pointers will always be different unless it is the same location

**Comment by timomeara** [comment]
is this going to be merged?

**Comment by AVVS** [comment]
Yes, we are working out a few cases not covered here & docs before getting this in

**Pull request merged**